client/layout/masterbar/quick-language-switcher.jsx

```diff
@@ -0,0 +1,59 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { getLocaleSlug } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+
+import MasterbarItem from './item';
+import LanguagePickerModal from 'components/language-picker/modal';
+import switchLocale from 'lib/i18n-utils/switch-locale';
+import config from 'config';
+
+class QuickLanguageSwitcher extends React.Component {
+	state = {
+		isShowingModal: false,
+	};
+
+	toggleLanguagesModal = () => {
+		this.setState( state => ( {
+			isShowingModal: ! state.isShowingModal,
+		} ) );
+	};
```

Comment:
Any reason why we're deferring assignment here?

Why not:

`toggleLanguagesModal = () => this.setState( { isShowingModal: ! this.state.isShowingModal } );`

?